### DISCUSSION TITLE
Shared eventEmitter

### DISCUSSION THREAD

**Discussion by igorjacauna** [bbc/peaks.js#503] [Q&A] [answered]
I'm building a multitrack app, and I wish to known if is it possible to have a shared EventEmitter to centralize the time update and other actions for each instance of Peaks.js

Maybe passing in options:

```ts
import EventEmitter from 'eventemitter3';

const myEventEmitter = new EventEmitter();

const myPlayer = new MyPlayer(myEventEmitter);


// And for each track in Multitrack that will belongs to a Peaks instance
Peaks.init({
  eventEmitter: myEventEmitter,
  player: myPlayer
});
```

I had think in make a PR but I thought that was better before to ask if is it possible or have any limitation to implement it

**Comment by chrisn** [comment]
This sounds interesting... there was a previous discussion in https://github.com/bbc/peaks.js/issues/432. Are you using multiple Peaks instances, for each track? I wonder if it's possible to ensure they're synchronized by using a shared `player` object. I suggest trying that before considering a shared EventEmitter - it could be that some events still need to be specific to each Peaks instance.

**Reply by igorjacauna** [reply]
Thanks for the fast answer!

Each track have one Peaks instance.

I tried the shared player, but seems that only the last one track added on stack was getting the `player.timeupdate`, resulting with a playhead frozen on begin in others tracks.

**Reply by igorjacauna** [reply]
I just created this [little project](https://github.com/igorjacauna/peaks-multitrack-poc) to help the discussion about multitrack and to show how only the last instance is getting the playhead moving when playing the tracks

Just run `yarn` and `yarn dev`

The pause is not working, sorry... 😓

**Reply by chrisn** [reply]
Thank you, that's really helpful. I'll take a look.

**Reply by thom4parisot** [reply]
If I understand well, the player provides independant interaction controls to the same audio source. And because Peaks reacts to standard HTML5 audio events, the player acts as a separate interface which is reflected as waveform updates as well.

**Reply by chrisn** [reply]
Yes, the Player interface is there to decouple the Peaks.js UI objects from the media player. The UI responds to `player.*` events emitted by the player, but otherwise doesn't know whether it's using a WebAudio `AudioContext` or an `<audio>` or `<video>` element or something else.

**Reply by chrisn** [reply, marked as answer]
I made [this example](https://gist.github.com/chrisn/376c73de59b1bcacb6f44bba45ea2811) to show how it might work. This is based on the [external player](https://github.com/bbc/peaks.js/blob/master/demo/external-player.html) demo page, so uses Tone.js.

**Reply by igorjacauna** [reply]
Thanks @chrisn, I will try with Tone.js and come back here with results soon

**Reply by igorjacauna** [reply]
Thats awesome @chrisn , it's working like a charm!

I updated the [little project ](https://github.com/igorjacauna/peaks-multitrack-poc) with this example

**Reply by chrisn** [reply]
Thanks! I'm just trying it out here with some multitrack recordings I found...
